compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7710,9 +7709,35 @@ public void visit(BLangTypeConversionExpr conversionExpr) {
     @Override
     public void visit(BLangLambdaFunction bLangLambdaFunction) {
         bLangLambdaFunction.function = rewrite(bLangLambdaFunction.function, bLangLambdaFunction.capturedClosureEnv);
+        BLangFunction function = bLangLambdaFunction.function;
+        // Add `@strand {thread: "any"}` annotation to an isolated named worker declaration in an isolated function.
+        if (function.flagSet.contains(Flag.WORKER) && Symbols.isFlagOn(function.symbol.type.flags, Flags.ISOLATED) &&
+                isInIsolatedFunction(env.enclInvokable)) {
+            addStrandAnnotationWithThreadAny(function.pos);
+            function.addAnnotationAttachment(this.strandAnnotAttachement);
+            BInvokableSymbol funcSymbol = function.symbol;
+            funcSymbol.addAnnotation(this.strandAnnotAttachement.annotationAttachmentSymbol);
+            funcSymbol.schedulerPolicy = SchedulerPolicy.ANY;
+        }
         result = bLangLambdaFunction;
     }
 
+    private boolean isInIsolatedFunction(BLangInvokableNode enclInvokable) {
+        if (enclInvokable == null) {
+            if (env.node.getKind() != NodeKind.EXPR_FUNCTION_BODY ||
+                    env.enclEnv.node.getKind() != NodeKind.ARROW_EXPR) {
+                return false;
+            }
+            return isIsolated(((BLangArrowFunction) env.enclEnv.node).funcType.flags);
```

Comment:
Can't this cast be cause CCE?

Reply:
used the same function in isolationAnalyzer. Willl check and update if necessary.